File: OrchidCore/src/test/java/com/eden/orchid/testhelpers/TestGeneratorModule.java
package com.eden.orchid.testhelpers;

import com.eden.orchid.api.generators.OrchidGenerator;
import com.eden.orchid.api.registration.OrchidModule;
import com.eden.orchid.impl.generators.HomepageGenerator;

public class TestGeneratorModule extends OrchidModule {

    private final Class<? extends OrchidGenerator> generatorClass;

    public TestGeneratorModule() {
        this(HomepageGenerator.class);
    }

    public TestGeneratorModule(Class<? extends OrchidGenerator> generatorClass) {
        this.generatorClass = generatorClass;
    }

    @Override
    protected void configure() {
        addToSet(OrchidGenerator.class, generatorClass);
    }

}

File: OrchidTest/src/main/kotlin/com/eden/orchid/testhelpers/TestGeneratorModule.kt
package com.eden.orchid.testhelpers

import com.eden.orchid.api.generators.OrchidGenerator
import com.eden.orchid.api.registration.OrchidModule

class TestGeneratorModule(
    private val generatorClass: Class<out OrchidGenerator<*>>
) : OrchidModule() {

    override fun configure() {
        addToSet(OrchidGenerator::class.java, generatorClass)
    }
}

inline fun <reified T : OrchidGenerator<*>> withGenerator(): OrchidModule {
    return TestGeneratorModule(T::class.java)
}